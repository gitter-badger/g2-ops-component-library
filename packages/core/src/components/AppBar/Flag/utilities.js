import { COUNTRY_CODES, FLAG_URL_PREFIX, HEIGHT_22PX, HEIGHT_30PX } from './consts'

const code = (countryCode: string): string => {
    return COUNTRY_CODES[countryCode] || countryCode
}
  
export const flagUrl = (countryCode: string): string => {
    return `${FLAG_URL_PREFIX}${code(countryCode)}.png`
}

export const imageClass = (type: string) => {
    return type === 'cas'
        ? HEIGHT_22PX
        : HEIGHT_30PX;
}