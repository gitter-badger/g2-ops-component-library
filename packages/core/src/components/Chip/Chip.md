Chip Example:

Useful for implementing Search chips and Filter chips that are currently used in Ops-portal.

```js
var chipData = [
  {key: 0, label: 'ReactJs'},
  {key: 1, label: 'JavaScript'},
  {key: 2, label: 'Material UI'},
  {key: 3, label: 'Fabric'},
];

<div style={{ display: 'flex', flexWrap: 'wrap' }}>
  {chipData.map((data) => (
    <Chip
      key={data.key}
      onRequestDelete={() => console.log('deletes the chip')}
      style={{ marginLeft: 4 }}
		>
      {data.label}
    </Chip>
  ))}
</div>
```