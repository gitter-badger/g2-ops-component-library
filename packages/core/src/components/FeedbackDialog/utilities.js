import { filter, isNil } from 'ramda'

import { SUBJECT_HEADER } from './consts';

export const createSubjectText = (target) => {
	return `${SUBJECT_HEADER} Feedback - Process: ${target.process} - Issue Type: ${target.issueType}`
}

// NOTE: Cloned from ops-portal repo.
export const generateFeedbackEmail = (props) => {
  return `
    Feedback: ${props.feedback}
    <br/>
    Email: ${props.email}
    <br/>
    Url: ${location.href}
    <br/>
    Selected Yard: ${props.selectedYard}
    <br/>
    Home Yard: ${props.homeYard}
    <br/>
    User Agent: ${navigator.userAgent}
  `
}

// NOTE: Cloned from ops-portal repo.
export const copyToClipboard = (text) => {
  const textField = document.createElement('textarea')
  textField.innerText = text
  document.body.appendChild(textField)
  textField.select()
  document.execCommand('copy')
  textField.remove()
}

// TODO: Clean up.
export const toObject = (form) => {
  if (!form || form.nodeName !== 'FORM') {
    return {}
  }
  const obj = {}
  let j = 0
  for (let i = form.elements.length - 1; i >= 0; i--) {
    if (form.elements[i].name === '') {
      continue
    }
    switch (form.elements[i].nodeName) {
      case 'INPUT':
        switch (form.elements[i].type) {
          case 'text':
          case 'email':
          case 'hidden':
          case 'password':
          case 'button':
          case 'reset':
          case 'submit':
            obj[form.elements[i].name] = form.elements[i].value
            break
          case 'checkbox':
          case 'radio':
            if (form.elements[i].checked) {
              obj[form.elements[i].name] = form.elements[i].value
            }
            break
          default:
            break
        }
        break
      case 'TEXTAREA':
        obj[form.elements[i].name] = form.elements[i].value
        break
      case 'SELECT':
        switch (form.elements[i].type) {
          case 'select-one':
            obj[form.elements[i].name] = form.elements[i].value
            break
          case 'select-multiple':
            for (j = form.elements[i].options.length - 1; j >= 0; j -= 1) {
              if (form.elements[i].options[j].selected) {
                obj[form.elements[i].name] = form.elements[i].options[j].value
              }
            }
            break
          default:
            break
        }
        break
      case 'BUTTON':
        switch (form.elements[i].type) {
          case 'reset':
          case 'submit':
          case 'button':
            obj[form.elements[i].name] = form.elements[i].value
            break
          default:
            break
        }
        break
      default:
        break
    }
  }
  return obj
}

// TODO: Clean up.
export const objectToQuery = (obj) => {
  const parts = []
  for (const i in obj) {
    if (obj.hasOwnProperty(i)) {
      if (Array.isArray(obj[i])) {
        [...new Set(obj[i])].map((val) => parts.push(`${encodeURIComponent(i)}=${encodeURIComponent(val)}`))
      } else {
        parts.push(`${encodeURIComponent(i)}=${encodeURIComponent(obj[i])}`)
      }
    }
  }
  return parts.join('&')
}

export const serialize = (form) => {
  const obj = toObject(form)
  return objectToQuery(obj)
}

const isClean = (x) => !(isNil(x) || x === '')
export const cleanData = (data) => filter(isClean, data)

// TODO: Clean up.
export const removeEmpty = (obj) => {
  const nonEmptyObj = {}
  for (const i in obj) {
    if (obj.hasOwnProperty(i) && obj[i] && !(obj[i] === 'A')) {
      nonEmptyObj[i] = obj[i]
    }
  }
  return nonEmptyObj
}

// TODO: Clean up.
export const dotToObj = (obj) => {
  const newObj = {}
  for (const k in obj) {
    if (obj.hasOwnProperty(k)) {
      const parts = k.split('.')
      const parentKey = parts.shift()
      newObj[parentKey] = parts.reverse().reduce((o, i) => {
        const res = {}
        res[i] = o
        return res
      }, obj[k])
    }
  }
  return newObj
}