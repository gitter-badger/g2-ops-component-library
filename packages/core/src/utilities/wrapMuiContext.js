// @flow
import type { ComponentType, Node } from 'react'

import React, { Component } from 'react'
import injectTapEventPlugin from 'react-tap-event-plugin'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import getMuiTheme from 'material-ui/styles/getMuiTheme'
import { initializeIcons } from '@uifabric/icons'

type ThemeType = {
  [string]: any,
}

const muiTheme = getMuiTheme()
const copartBlue = '#1d5ab9'
const niceBlue = '#4a90e2'

export const copartBaseTheme: ThemeType = {
    ...muiTheme,
  appBar: {
    ...muiTheme.appBar,
    color: copartBlue,
  },
  baseTheme: {
    ...muiTheme.baseTheme,
    palette: {
      ...muiTheme.baseTheme.palette,
      pickerHeaderColor: copartBlue,
      primary1Color: copartBlue,
      primary2Color: copartBlue,
    },
  },
  flatButton: {
    ...muiTheme.flatButton,
    primaryTextColor: copartBlue,
  },
  palette: {
    ...muiTheme.palette,
    pickerHeaderColor: copartBlue,
    primary1Color: copartBlue,
    primary2Color: copartBlue,
  },
  datePicker: {
    ...muiTheme.datePicker,
    headerColor: copartBlue,
    color: copartBlue,
    selectColor: copartBlue,
  },
  toggle: {
    ...muiTheme.toggle,
    thumbOnColor: copartBlue,
    trackOnColor: niceBlue,
  },
  timePicker: {
    ...muiTheme.timePicker,
    headerColor: copartBlue,
    accentColor: copartBlue,
    selectColor: copartBlue,
  },
  stepper: {
    ...muiTheme.stepper,
    iconColor: copartBlue,
  },
  tabs: {
    selectedTextColor: 'rgba(38, 166, 91, 1.0)',
    textColor: 'rgba(218, 223, 225, 1.0)',
  },
  inkBar: {
    backgroundColor: '#545A63',
  },
}

initializeIcons()
injectTapEventPlugin()

// a higher order function that provides mui context to the Component being passed.
export function wrapMuiContext<T>(WrapperComponent: ComponentType<T>): (T) => Node {
  return (props) => (
    <MuiThemeProvider muiTheme={copartBaseTheme}>
      <WrapperComponent {...props} />
    </MuiThemeProvider>
  )
}
