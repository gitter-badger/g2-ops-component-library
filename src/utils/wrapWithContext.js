import React from 'react'
import WithContext from 'react-with-context'
import baseTheme from 'material-ui/styles/baseThemes/lightBaseTheme'
import createMuiTheme from 'material-ui/styles/getMuiTheme'

const wrapWithContext = (component, context) => <WithContext context={context}>{component}</WithContext>

export const theme = createMuiTheme(baseTheme)

export const wrapWithMaterialUIContext = (component) =>
  wrapWithContext(component, { muiTheme: createMuiTheme(baseTheme) })

export default wrapWithContext
