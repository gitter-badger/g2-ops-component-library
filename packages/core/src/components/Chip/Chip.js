// @flow

import type { Node, ChildrenArray } from 'react'

import React from 'react'
import MuiChip from 'material-ui/Chip'
import { wrapMuiContext } from 'utilities/wrapMuiContext'

type StyleType = {
  [string]: mixed,
}

type ChippropTypes = {
  /** Override the background color of the chip. */
  backgroundColor: string,
  /** Used to render elements inside the Chip. */
  children?: ChildrenArray<Node>,
  /** CSS className of the root element. */
  className: string,
  /** Override the label color. */
  labelColor: string,
  /** Override the inline-styles of the label. */
  labelStyle: StyleType,
  /** Callback function fired when the delete icon is clicked. If set, the delete icon will be shown. */
  onRequestDelete: (SyntheticMouseEvent<HTMLInputElement>) => void,
  /** Callback function fired when the Chip element is touch-tapped. */
  onClick: (SyntheticMouseEvent<HTMLInputElement>) => void,
  /** Override the inline-styles of the root element. */
  style: StyleType,
}

export const Chip = wrapMuiContext(({
  backgroundColor,
  children,
  className,
  labelColor,
  labelStyle,
  onRequestDelete,
  onClick,
  style,
}: ChippropTypes) => (
  <MuiChip
    backgroundColor={backgroundColor}
    className={className}
    labelColor={labelColor}
    labelStyle={labelStyle}
    onRequestDelete={onRequestDelete}
    onClick={onClick}
    style={style}
  >
    {children}
  </MuiChip>
))
