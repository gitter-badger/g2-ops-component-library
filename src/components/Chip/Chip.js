// @flow

import type { Node, ChildrenArray } from 'react'

import React from 'react'
import MuiChip from 'material-ui/Chip'
import { wrapMuiContext } from '../../wrapMuiContext'

type StyleType = {
  [string]: mixed,
}

type ChippropTypes = {
  /** Override the background color of the chip. */
  backgroundColor: string,
  /** Used to render elements inside the Chip. */
  children?: ChildrenArray<Node>,
  /** CSS className of the root element. */
  className: string,
  /** Override the label color. */
  labelColor: string,
  /** Override the inline-styles of the label. */
  labelStyle: StyleType,
  /** Callback function fired when the delete icon is clicked. If set, the delete icon will be shown. */
  onRequestDelete: (SyntheticMouseEvent<HTMLInputElement>) => void,
  /** Callback function fired when the Chip element is touch-tapped. */
  onTouchTap: (SyntheticMouseEvent<HTMLInputElement>) => void,
  /** Override the inline-styles of the root element. */
  style: StyleType,
}

const Chip = ({
  backgroundColor,
  children,
  className,
  labelColor,
  labelStyle,
  onRequestDelete,
  onTouchTap,
  style,
}: ChippropTypes) => (
  <MuiChip
    backgroundColor={backgroundColor}
    className={className}
    labelColor={labelColor}
    labelStyle={labelStyle}
    onRequestDelete={onRequestDelete}
    onTouchTap={onTouchTap}
    style={style}
  >
    {children}
  </MuiChip>
)

export default wrapMuiContext(Chip)
