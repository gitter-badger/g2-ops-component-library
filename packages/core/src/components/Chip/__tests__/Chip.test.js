import React from 'react'
import renderer from 'react-test-renderer'

import { wrapWithMaterialUIContext } from 'utilities/wrapWithContext'

import {Chip} from '../Chip'

describe('<Chip />', () => {
  test('should render properly', () => {
    const tree = renderer.create(wrapWithMaterialUIContext(<Chip>Sample Chip</Chip>))
    expect(tree).toMatchSnapshot()
  })
})
