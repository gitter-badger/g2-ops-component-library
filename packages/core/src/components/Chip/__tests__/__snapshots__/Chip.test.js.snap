// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chip /> should render properly 1`] = `
<div
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
      "backgroundColor": "rgb(224, 224, 224)",
      "border": 10,
      "borderRadius": 16,
      "boxShadow": null,
      "boxSizing": "border-box",
      "cursor": "default",
      "display": "flex",
      "fontFamily": "Roboto, sans-serif",
      "fontSize": "inherit",
      "fontWeight": "inherit",
      "margin": 0,
      "muiPrepared": true,
      "outline": "none",
      "padding": 0,
      "position": "relative",
      "textDecoration": "none",
      "verticalAlign": null,
      "whiteSpace": "nowrap",
      "width": "fit-content",
    }
  }
  tabIndex={0}
>
  <span
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontSize": 14,
        "fontWeight": 400,
        "lineHeight": "32px",
        "muiPrepared": true,
        "paddingLeft": 12,
        "paddingRight": 12,
        "userSelect": "none",
        "whiteSpace": "nowrap",
      }
    }
  >
    Sample Chip
  </span>
</div>
`;
